### PR TITLE
Lesson2

### PR DESCRIPTION
**Pull request by aksafan**

### CODE REVIEW

File: sql2.txt [hunk 1]
```diff
@@ -2,44 +2,45 @@
 to join the OrderDetails table with the product table.  For TotalCount, you will Use
 SUM(Quantity) AS TotalCount .  Order by product name.
 You will need to group by product ID.  Paste your SQL statement below.
-
+SELECT p.ProductName,SUM(od.Quantity) as TotalCount FROM Products p JOIN OrderDetails od ON p.ProductID = od.ProductID GROUP BY p.ProductName;
```
Comment (josiahandrew12) [inline sql2.txt:5]: 👍

File: sql2.txt [hunk 2]
```diff
@@ -2,44 +2,45 @@
...
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
-
+SELECT c.CustomerName,p.ProductName,SUM(od.Quantity) AS TotalQuantity,SUM(od.Quantity * p.Price) AS TotalPrice FROM Customers c JOIN OrderDetails od ON c.CustomerID = o.CustomerID JOIN Orders o ON o.OrderID = od.OrderID JOIN Products p ON p.ProductID = od.ProductID GROUP BY c.CustomerName,p.ProductName;
 
 5. Add a new Order to the Orders table.  You will have to specify the ID of an existing customer
 to satisfy the foreign key constraint.  You will also have to specify the ID of an existing
 employee.  Paste your SQL statement below.
-
+INSERT INTO Orders (CustomerID,EmployeeID,OrderDate,ShipperID) VALUES (3,5,"7/11/1996",2);
```
Comment (josiahandrew12) [inline sql2.txt:30]: Nice job here but I think usually the correct format for Date would be` 1996-07-11`

File: sql2.txt [hunk 3]
```diff
@@ -2,44 +2,45 @@
...
 
 6. Add three OrderDetails.  These should specify the ID of the new Order you just created.  For each,
 you will have to specify the ID of an existing product.  Paste your SQL statement below.
+INSERT INTO OrderDetails (OrderID,ProductID,Quantity) VALUES (10444,3,5),(10444,4,10),(10444,5,1);
```
Comment (josiahandrew12) [inline sql2.txt:34]: Functionality is correct just need some spaces between  `(OrderID, ProductID, Quantity)`

File: sql2.txt [hunk 4]
```diff
@@ -49,4 +50,13 @@ DELETE FROM OrderDetails WHERE OrderID IN (SELECT OrderID FROM Orders WHERE Cust
 
 What are the other two statements?  Try them out, and then paste them below.
 
+I've just deleted Customer 76 with `DELETE FROM Customers WHERE CustomerID = 76;` and it is no more exists.
+I suppose the question was about foreign keys restriction policy, but I've checked it for Customers with 
+`SELECT * FROM (SELECT m.name, p.* FROM sqlite_master m JOIN pragma_foreign_key_list(m.name) p ON m.name != p."table" WHERE m.type = 'table' ORDER BY m.name) as t WHERE t."table" = 'Customers';`
+and there is `NO ACTION` for both `on_update` and `on_delete`.
+
+This being said, based on accessable info, the 2nd query should be:
```
Comment (josiahandrew12) [inline sql2.txt:58]: 👍

**Review by josiahandrew12 — APPROVED** [review]

File: sql2.txt (hunk 2 above)
Comment (aksafan) [reply sql2.txt:30]: Usually It depends on a situation and agreements in a team. Moreover, date as int is more preferable as one can avoid time zones issues with storing date as an integer.

Here this format `"7/11/1996"` was chosen only cause all other records in this table were in this exact format.

Comment (josiahandrew12) [reply sql2.txt:30]: Ahh my mistake  actually meant the string  "1996-07-11" with quotes forgot to add them but yeah if the format already set no problem.

**Pull request merged**